tests/integration/simple_integration_test.go

```diff
@@ -0,0 +1,164 @@
+			clientset, err := k8s.GetKubernetesClientFromOptionsE(GinkgoT(), kubectlOptions)
+			pods, _ := clientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "name=cass-operator"})
+
+			Expect(len(pods.Items)).To(Equal(1))
+			k8s.WaitUntilPodAvailable(GinkgoT(), kubectlOptions, pods.Items[0].Name, 50, 500 * time.Millisecond)
+
+			// We could use also kubectl wait --for=condition=Ready pod -l name=cass-operator
+
+			// Deploy traefik
+			// helm install traefik traefik/traefik -n traefik --create-namespace -f docs/content/en/docs/topics/ingress/traefik/kind-deployment/traefik.values.yaml
+			//valuesPath, _ := filepath.Abs("../../docs/content/en/docs/topics/ingress/traefik/kind-deployment/traefik.values.yaml")
+			//helm.RunHelmCommandAndGetOutputE(GinkgoT(), options, "install", "traefik", "traefik/traefik", "-n", "traefik", "--create-namespace", "-f", valuesPath)
+
+			// Deploy the cluster
+			clusterChartPath, err := filepath.Abs("../../charts/k8ssandra-cluster")
+			Expect(err).To(BeNil())
+
+			releaseNameCluster = releaseName + "-cluster"
+			options = &helm.Options{
+				// Enable traefik to allow redirections for testing
+				SetValues: map[string]string{
+					"ingress.traefik.enabled": "true",
+					"ingress.traefik.monitoring.grafana.host": "grafana.localhost",
+					"ingress.traefik.monitoring.prometheus.host": "prometheus.localhost",
+					"ingress.traefik.repair.host": "repair.localhost",
+				},
+				KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+			}
+			helm.Install(GinkgoT(), options, clusterChartPath, releaseNameCluster)
+
+			// Wait for CassandraDatacenter to be ready..
+			k8s.RunKubectl(GinkgoT(), kubectlOptions, "wait", "--for=condition=Ready", "cassandradatacenter/dc1", "--timeout=300s")
+
+			// Verify all the correct services are there
+			// Grafana
+			service := k8s.GetService(GinkgoT(), kubectlOptions, "grafana-service")
+			Expect(service).ToNot(BeNil())
+
+			// Prometheus
+			service = k8s.GetService(GinkgoT(), kubectlOptions, "prometheus-operated")
+			Expect(service).ToNot(BeNil())
+
+			// Cassandra
+			service = k8s.GetService(GinkgoT(), kubectlOptions, "k8ssandra-dc1-service")
+			Expect(service).ToNot(BeNil())
+			service = k8s.GetService(GinkgoT(), kubectlOptions, "k8ssandra-dc1-all-pods-service")
+			Expect(service).ToNot(BeNil())
+			service = k8s.GetService(GinkgoT(), kubectlOptions, "k8ssandra-seed-service")
+			Expect(service).ToNot(BeNil())
+
+			// Reaper
+			// Medusa (if enabled)
+
+			// Verify traefik is ready
+			kubectlOptions = k8s.NewKubectlOptions("", "", "traefik")
+			k8s.RunKubectl(GinkgoT(), kubectlOptions, "wait", "--for=condition=Ready", "pod", "-l", "app.kubernetes.io/name=traefik", "--timeout=300s")
+			kubectlOptions = k8s.NewKubectlOptions("", "", namespace)
+
+			// TODO Verify with cqlsh that Cassandra is working properly
+
+			// Verify that prometheus is polling the Cassandra instance correctly
```

Comment:
I really like the approach you are taking with the Prometheus verification! This would catch things like screwing up the ServiceMonitor or if the metric collector is not deployed with Cassandra.